### PR TITLE
Adds mentor tickets + Makes admin tickets nicer to look at

### PR DESCRIPTION
**Pull request by Kmc2000**
Basically a dumbed down version of:
https://github.com/DDMers/Deep-Space-13/pull/134

Mentors retain admin status. We can mess with that via configs.

Upon ticketing you now see this:

Which sorts your ahelp/mhelp via a special colour and a special super secret sound effect.

:cl: Kmc2000
add: Mentor tickets now exist. Use them for gameplay questions!
/:cl:

Thanks to crossed for porting hippie's one anyway, but this is cleaner and far easier to manage with fewer changes necessary.

### CODE REVIEW

File: code/modules/admin/verbs/adminhelp.dm [hunk 1]
```diff
@@ -513,8 +539,10 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 		else
 			current_ticket.AddInteraction("[key_name_admin(usr)] opened a new ticket.")
 			current_ticket.Close()
-
-	new /datum/admin_help(msg, src, FALSE)
+	if(alert(usr,"Is this an administrative issue, or a gameplay question?",,"Admin issue","Question") == "Question")
+		new /datum/admin_help(msg, src, FALSE, TRUE)
```
Comment (alexkar598) [inline code/modules/admin/verbs/adminhelp.dm]: For the love of god use named arguments for the boolean flags
```suggestion
		new /datum/admin_help(msg, src, is_bwoink = FALSE,mentor_status = TRUE)
```

File: code/modules/admin/verbs/adminhelp.dm [hunk 2]
```diff
@@ -513,8 +539,10 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 		else
 			current_ticket.AddInteraction("[key_name_admin(usr)] opened a new ticket.")
 			current_ticket.Close()
-
-	new /datum/admin_help(msg, src, FALSE)
+	if(alert(usr,"Is this an administrative issue, or a gameplay question?",,"Admin issue","Question") == "Question")
+		new /datum/admin_help(msg, src, FALSE, TRUE)
+	else
+		new /datum/admin_help(msg, src, FALSE, FALSE)
```
Comment (alexkar598) [inline code/modules/admin/verbs/adminhelp.dm]: .

File: code/modules/admin/verbs/adminhelp.dm [hunk 3]
```diff
@@ -513,8 +539,10 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 		else
 			current_ticket.AddInteraction("[key_name_admin(usr)] opened a new ticket.")
 			current_ticket.Close()
-
-	new /datum/admin_help(msg, src, FALSE)
+	if(alert(usr,"Is this an administrative issue, or a gameplay question?",,"Admin issue","Question") == "Question")
```
Comment (alexkar598) [inline code/modules/admin/verbs/adminhelp.dm]: Mention in the question if its a mentor help or an admin help
```suggestion
	if(alert(usr,"Is this an administrative issue, or a gameplay question?",,"Admin issue (adminhelp)","Question (mentorhelp)") == "Question")
```

File: code/modules/admin/verbs/adminhelp.dm [hunk 4]
```diff
@@ -387,23 +411,23 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
...
-	dat += "<h4>Admin Help Ticket #[id]: [LinkedReplyName(ref_src)]</h4>"
+	dat += "<h4>Ticket #[id]: [LinkedReplyName(ref_src)]</h4>"
 	dat += "<b>State: "
 	switch(state)
 		if(AHELP_ACTIVE)
-			dat += "<font color='red'>OPEN</font>"
+			dat += "<font color='#ff8c8c'>OPEN</font>"
 		if(AHELP_RESOLVED)
-			dat += "<font color='green'>RESOLVED</font>"
+			dat += "<font color='#9adb92'>RESOLVED</font>"
```
Comment (alexkar598) [inline code/modules/admin/verbs/adminhelp.dm:420]: Is the color difference so important we should sacrifice readability for it?

File: code/modules/admin/verbs/adminhelp.dm [hunk 5]
```diff
@@ -387,23 +411,23 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 /datum/admin_help/proc/TicketPanel()
 	var/list/dat = list("<html><head><title>Ticket #[id]</title></head>")
 	var/ref_src = "[REF(src)]"
-	dat += "<h4>Admin Help Ticket #[id]: [LinkedReplyName(ref_src)]</h4>"
+	dat += "<h4>Ticket #[id]: [LinkedReplyName(ref_src)]</h4>"
 	dat += "<b>State: "
 	switch(state)
 		if(AHELP_ACTIVE)
-			dat += "<font color='red'>OPEN</font>"
+			dat += "<font color='#ff8c8c'>OPEN</font>"
```
Comment (alexkar598) [inline code/modules/admin/verbs/adminhelp.dm:418]: Is the color difference so important we should sacrifice readability for it?

File: code/modules/admin/verbs/adminhelp.dm [hunk 6]
```diff
@@ -163,7 +173,7 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
 //call this on its own to create a ticket, don't manually assign current_ticket
 //msg is the title of the ticket: usually the ahelp text
 //is_bwoink is TRUE if this ticket was started by an admin PM
-/datum/admin_help/New(msg, client/C, is_bwoink)
+/datum/admin_help/New(msg, client/C, is_bwoink, mentor_status)
```
Comment (alexkar598) [inline code/modules/admin/verbs/adminhelp.dm]: use default arguments for mentor_status
```suggestion
/datum/admin_help/New(msg, client/C, is_bwoink, mentor_status = FALSE)
```

File: code/modules/admin/verbs/adminhelp.dm [hunk 7]
```diff
@@ -188,14 +198,19 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
...
 		message_admins("<span class='notice'>Ticket [TicketHref("#[id]")] created</span>")
+
 	else
 		MessageNoRecipient(msg)
-
-		SSredbot.send_discord_message("admin", "Ticket #[id] created by [usr.ckey] ([usr.real_name]): [name]", "ticket") //send the ticket to discord
+		if(is_mentor_ticket)
+			SSredbot.send_discord_message("admin", "Mentor ticket #[id] created by [usr.ckey] ([usr.real_name]): [name]", "ticket") //send the ticket to discord
+		else
+			SSredbot.send_discord_message("admin", "Admin Ticket #[id] created by [usr.ckey] ([usr.real_name]): [name]", "ticket") //send the ticket to discord
```
Comment (alexkar598) [inline code/modules/admin/verbs/adminhelp.dm:213]: @crossedfall mind confirmed this wont break anything with the bot?

File: code/modules/admin/verbs/adminhelp.dm [hunk 8]
```diff
@@ -256,19 +271,28 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
...
+	AddInteraction("<font color='#ff8c8c'>[LinkedReplyName(ref_src)]: [msg]</font>")
 
 	//send this msg to all admins
 	for(var/client/X in GLOB.admins)
 		if(X.prefs.toggles & SOUND_ADMINHELP)
-			SEND_SOUND(X, sound('sound/effects/adminhelp.ogg'))
+			if(is_mentor_ticket) //DeepSpace13 - Mentors
+				SEND_SOUND(X, sound('DS13/sound/effects/mentorhelp.ogg'))
+			else
+				SEND_SOUND(X, sound('sound/effects/adminhelp.ogg'))
```
Comment (alexkar598) [inline code/modules/admin/verbs/adminhelp.dm:287]: This changes it for the admin bwoink but not for the bwoink the player gets when he has an answer to his mentor help

**Review by alexkar598 — CHANGES_REQUESTED** [review]

File: code/modules/admin/verbs/adminhelp.dm (hunk 2 above)
Comment (francinum) [reply code/modules/admin/verbs/adminhelp.dm]: These should just be different verbs anyways. No reason to combine them.

File: code/modules/admin/verbs/adminhelp.dm (hunk 7 above)
Comment (Crossedfall) [reply code/modules/admin/verbs/adminhelp.dm:213]: I confirm it wont break anything. That being said, I have to ask if you actually want mentor tickets to be sent to discord in the same way as ahelps.

File: code/modules/admin/verbs/adminhelp.dm (hunk 4 above)
Comment (Crossedfall) [reply code/modules/admin/verbs/adminhelp.dm:420]: You either sacrifice readability here or in the UI. Given how ubiquitous hex colors are, I don't honestly think this is a problem.

File: code/modules/admin/verbs/adminhelp.dm (hunk 5 above)
Comment (Crossedfall) [reply code/modules/admin/verbs/adminhelp.dm:418]: You either sacrifice readability here or in the UI. Given how ubiquitous hex colors are, I don't honestly think this is a problem.

File: code/modules/admin/verbs/adminhelp.dm (hunk 7 above)
Comment (Kmc2000) [reply code/modules/admin/verbs/adminhelp.dm:213]: I think it's valuable for mentors to see if there are any gameplay questions they can easily answer

File: code/modules/admin/verbs/adminhelp.dm (hunk 8 above)
Comment (Kmc2000) [reply code/modules/admin/verbs/adminhelp.dm:287]: Yeah hearing data's voice (the mentorhelp sound: "How may I assist you?") would get tiresome. Perhaps another sound is in order?

Comment (alexkar598) [reply code/modules/admin/verbs/adminhelp.dm:287]: The same line but said by the Borg one time on two.

File: code/modules/admin/verbs/adminhelp.dm (hunk 5 above)
Comment (alexkar598) [reply code/modules/admin/verbs/adminhelp.dm:418]: Thing is doubt the UI inpact is high since it's a slight change in color tone

**Comment by Crossedfall** [comment]
Clearly not a front-end developer then :^)

File: code/modules/admin/verbs/adminhelp.dm (hunk 5 above)
Comment (Kmc2000) [reply code/modules/admin/verbs/adminhelp.dm:418]: It's unreadable without this

Comment (alexkar598) [reply code/modules/admin/verbs/adminhelp.dm:418]: Eh sure,whatever

File: code/modules/admin/verbs/adminhelp.dm (hunk 2 above)
Comment (alexkar598) [reply code/modules/admin/verbs/adminhelp.dm]: I'm really just here to tell people how to optimize their code not how to design their code.At the end of the day,it just depends on how much we want to trust mentors with admin info by giving them an admin datum(giving access to read-only VV and tons of round info) or making a new system of mentor datums just for them.

**Review by Crossedfall — COMMENTED** [review]
I would recommend making changes to `TimeoutVerb()`  so that general questions don't lock the verb for an extended period of time. Ideally you would have it timeout each option separately, since you could have a general question and then a legitimate ahelp need shortly after. Added bonus would be that you could configure the mentorHelps to have a shorter timeout period. 

Alternative suggestion: Only trigger the timeout if it's an ahelp.

File: code/modules/admin/verbs/adminhelp.dm [hunk 9]
```diff
@@ -188,14 +198,19 @@ GLOBAL_DATUM_INIT(ahelp_tickets, /datum/admin_help_tickets, new)
...
 	if(is_bwoink)
 		AddInteraction("<span class='notice'>[key_name_admin(usr)] PM'd [LinkedReplyName()]</span>")
 		message_admins("<span class='notice'>Ticket [TicketHref("#[id]")] created</span>")
+
 	else
 		MessageNoRecipient(msg)
-
-		SSredbot.send_discord_message("admin", "Ticket #[id] created by [usr.ckey] ([usr.real_name]): [name]", "ticket") //send the ticket to discord
+		if(is_mentor_ticket)
+			SSredbot.send_discord_message("admin", "Mentor ticket #[id] created by [usr.ckey] ([usr.real_name]): [name]", "ticket") //send the ticket to discord
```
Comment (Crossedfall) [inline code/modules/admin/verbs/adminhelp.dm]:
```suggestion
			SSredbot.send_discord_message("mentor", "Mentor ticket #[id] created by [usr.ckey] ([usr.real_name]): [name]", "ticket") //send the ticket to discord
```

New function added into the status cog to check for a `mentor` announcement channel. Mentor tickets will display with a different color embed and will not use `@here` pings if there aren't any admins online.

image

The mentor channel can be set with `[p]setstatus mentorchannel <channel>`  where [p] is your prefix.

See https://github.com/crossedfall/crossed-cogs/commit/1b3078476a2c4e61cb77f79f062fd22b6afd40b1 for changes.

File: code/modules/admin/verbs/adminhelp.dm (hunk 7 above)
Comment (crossedfall) [reply code/modules/admin/verbs/adminhelp.dm]: Changes made to the bot. See my new comment with suggested changes.

**Review by Crossedfall — CHANGES_REQUESTED** [review]

**Comment by francinum** [comment]
@Kmc2000 There appears to be no distinction between Admins and mentors here. This is an issue as there's an inherent separation between these two levels of trust and responsibility.

**Pull request merged**